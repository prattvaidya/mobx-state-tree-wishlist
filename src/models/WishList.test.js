import { WishList, WishListItem } from "./WishList";

it("can create an instance of a model", () => {
  const item = WishListItem.create({
    name: "The subtle art of not giving a f***",
    price: 29.99,
    // image:
    //   "https://upload.wikimedia.org/wikipedia/en/b/bd/The_Subtle_Art_of_Not_Giving_a_F%2Ack_by_Mark_Manson_-_Book_Cover.png",
  });

  expect(item.price).toBe(29.99);
  expect(item.image).toBe("");
});

it("can create a wishlist", () => {
  const list = WishList.create({
    items: [{ name: "The subtle art of not giving a f***", price: 29.99 }],
  });

  expect(list.items.length).toBe(1);
  expect(list.items[0].price).toBe(29.99);
});
