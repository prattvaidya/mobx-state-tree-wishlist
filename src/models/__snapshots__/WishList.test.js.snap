// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can add new items - 2 1`] = `
Array [
  Object {
    "op": "add",
    "path": "/items/0",
    "value": Object {
      "image": "",
      "name": "South of the Border, West of the Sun",
      "price": 25,
    },
  },
  Object {
    "op": "replace",
    "path": "/items/0/name",
    "value": "Maximum City",
  },
]
`;

exports[`can add new items 1`] = `
Object {
  "items": Array [
    Object {
      "image": "",
      "name": "Maximum City",
      "price": 25,
    },
  ],
}
`;

exports[`can add new items 2`] = `
Array [
  Object {
    "items": Array [
      Object {
        "image": "",
        "name": "South of the Border, West of the Sun",
        "price": 25,
      },
    ],
  },
  Object {
    "items": Array [
      Object {
        "image": "",
        "name": "Maximum City",
        "price": 25,
      },
    ],
  },
]
`;
